test/auditing.jl

```diff
@@ -788,57 +789,113 @@ end
+                    # This library has soft-float ABI
+                    echo 'int _start() { return 0; }' | /opt/${target}/bin/${target}-gcc -nostdlib -shared -mfloat-abi=soft -o "${libdir}/libsoft.${dlext}" -x c -
+                    # hello_world built by Go doesn't specify any float ABI
+                    make -C /usr/share/testsuite/go/hello_world/
+                    cp "/tmp/testsuite/${target}/go/hello_world/hello_world" "${bindir}/hello_world"
+                    """,
+                    # Build for Linux armv7l hard-float
+                    [platform],
+                    # Ensure our library product is built
+                    [
+                        LibraryProduct("libhard", :libhard),
+                        LibraryProduct("libsoft", :libsoft),
+                        ExecutableProduct("hello_world", :hello_world),
+                    ],
+                    # No dependencies
+                    Dependency[];
+                    compilers = [:c, :go],
+                    verbose = true,
+                    require_license = false
+                )
+            end
+
+            @test haskey(build_output_meta, platform)
+            tarball_path, tarball_hash = build_output_meta[platform][1:2]
+            @test isfile(tarball_path)
+
+            # Unpack it somewhere else
+            @test verify(tarball_path, tarball_hash)
+            testdir = joinpath(build_path, "testdir")
+            mkdir(testdir)
+            unpack(tarball_path, testdir)
+            # Remove libsoft.so, we want to run audit only on the other products
+            rm(joinpath(testdir, "lib", "libsoft.so"))
+            # Make sure `hello_world` passes the float ABI check even if it doesn't
+            # set `EF_ARM_ABI_FLOAT_HARD`.
+            @test Auditor.audit(Prefix(testdir); platform=platform, require_license=false)
         end
+    end
+    @testset "OS/ABI: $platform" for platform in [Platform("x86_64", "freebsd"),
+                                                  Platform("aarch64", "freebsd")]
+        mktempdir() do build_path
+            build_output_meta = @test_logs (:warn, r"libwrong.so has an ELF header OS/ABI value that is not set to FreeBSD") match_mode=:any begin
+                autobuild(
+                    build_path,
+                    "OSABI",
+                    v"4.2.0",
+                    # No sources
+                    FileSource[],
+                    # Build a library with a mismatched OS/ABI in the ELF header
+                    raw"""
+                    mkdir -p "${libdir}"
+                    echo 'int wrong() { return 0; }' | cc -shared -fPIC -o "${libdir}/libwrong.${dlext}" -x c -
+                    echo 'int right() { return 0; }' | cc -shared -fPIC -o "${libdir}/libright.${dlext}" -x c -
+                    # NetBSD runs anywhere, which implies that anything that runs is for NetBSD, right?
+                    patchelf --set-os-abi 2 "${libdir}/libwrong.${dlext}"
```

Comment:
```suggestion
                    raw"""
                    apk update
                    apk add binutils
                    mkdir -p "${libdir}"
                    echo 'int wrong() { return 0; }' | cc -shared -fPIC -o "${libdir}/libwrong.${dlext}" -x c -
                    echo 'int right() { return 0; }' | cc -shared -fPIC -o "${libdir}/libright.${dlext}" -x c -
                    # NetBSD runs anywhere, which implies that anything that runs is for NetBSD, right?
                    elfedit --output-osabi=NetBSD "${libdir}/libwrong.${dlext}"
```
I wonder if this will work